### PR TITLE
MOAI-879 (DIRS Rules updates)

### PR DESCRIPTION
**Pull request by mpartha-dl**
Created DIRS Rules 57.01.0001 and 57.01.0002

### CODE REVIEW

**Review by mmarable-dlp — APPROVED** [review]

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.01.0001.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  Descriptor.CodeValue = 'Weapon'
+
+   AND
+   NumWeapons IS NULL 
+)
+
+INSERT INTO 
+	rules.RuleValidationDetail (RuleValidationId, Id, RuleId, IsError, [Message])
+SELECT
+	@RuleValidationId, SchoolId, @RuleId RuleId, @IsError IsError, @Message [Message]
```
Comment (steve-dlp) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.01.0001.sql:43]: I wouldn't use Schoolid here - you have better options - StudentUniqueId is probably the best option.  So you'll have to update the select to pull the studentuniqueid.

Also I would include additional data in the message - like the incident identifier, the weapon descriptor **code value**, and the school id.

File: ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.01.0002.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    OR Descriptor.CodeValue = 'Hand Gun')
+
+  AND NumWeapons <> 1
+  )
+ 
+
+INSERT INTO 
+	rules.RuleValidationDetail (RuleValidationId, Id, RuleId, IsError, [Message])
+SELECT
+	@RuleValidationId, SchoolId, @RuleId RuleId, @IsError IsError, @Message [Message]
```
Comment (steve-dlp) [inline ValidationWeb/Content/SqlRules/DIRSRuleSet/dirs-rule-57.01.0002.sql:39]: Again - here you should use the studentuniqueid.  Whenever your rule can be related to a student - you'll want to insert the studentuniqueid in the rulevalidationdetail.Id column.
You should also provide additional info to help the user in the message such as the Incident Identifier and the schoolid.

**Review by steve-dlp — CHANGES_REQUESTED** [review]
see comments @mpartha-dl

**Comment by steve-dlp** [comment]
These rules need a @districtid filter 
@mmarable-dlp can you adjust please?

**Review by mmarable-dlp — APPROVED** [review]

**Review by steve-dlp — APPROVED** [review]

**Pull request merged**